syncd/syncd_cli.cpp

```diff
@@ -0,0 +1,269 @@
+
+int cli_server_socket = 0;
+int cli_client_socket = 0;
+
+bool sendtoclient(const std::string& s)
+{
+    SWSS_LOG_ENTER();
+
+    if (write(cli_client_socket, s.c_str(), s.length()) < 0)
+    {
+        SWSS_LOG_ERROR("write failed, errno: %s", strerror(errno));
+        return false;
+    }
+
+    return true;
+}
+
+void cmd_exit()
+{
+    SWSS_LOG_ENTER();
+
+    run_user = false;
+
+    sendtoclient("bye\n");
+}
+
+void cmd_help(const std::vector<std::string>& args)
+{
+    SWSS_LOG_ENTER();
+
+    std::string help = "\n\
+    help            - this command\n\
+    exit            - close cli connection\n\
+    loglevel LEVEL  - sets debug logger level\n\n";
+
+    sendtoclient(help);
+}
+
+void cmd_loglevel(const std::vector<std::string>& args)
+{
+    SWSS_LOG_ENTER();
+
+    if (args.size() < 2)
+    {
+        sendtoclient("missing level parameter\n");
+        return;
+    }
+
+    std::string slevel = args[1];
+
+    std::transform(slevel.begin(), slevel.end(), slevel.begin(), ::toupper);
+
+    swss::Logger::Priority level = swss::Logger::stringToPriority(slevel);
+
+    swss::Logger::getInstance().setMinPrio(level);
+
+    sendtoclient("log level changed to " + swss::Logger::priorityToString(level) + "\n");
+}
+
+std::string trim(const std::string &in)
+{
+    std::string s = in;
+
+    s.erase(std::find_if(s.rbegin(), s.rend(),
+                std::not1(std::ptr_fun<int, int>(std::isspace))).base(), s.end());
+
+    s.erase(s.begin(), std::find_if(s.begin(), s.end(),
+                std::not1(std::ptr_fun<int, int>(std::isspace))));
+
+    return s;
+}
+
+void handle_cli_commands(const std::string& line)
+{
+    SWSS_LOG_ENTER();
+
+    SWSS_LOG_NOTICE("cli: %s", line.c_str());
+
+    // TODO this must be more smart, to split multiple spaces
+    auto tokens = swss::tokenize(trim(line), ' ');
+
+    if (tokens.size() < 1)
+    {
+        return;
+    }
+
+    const std::string& cmd = tokens[0];
+
+    if (cmd == "exit" || cmd == "quit")
+    {
+        cmd_exit();
+    }
+    else if (cmd == "loglevel")
+    {
+        cmd_loglevel(tokens);
+    }
+    else if (cmd == "help")
+    {
+        cmd_help(tokens);
+    }
+    else
+    {
+        sendtoclient("unknown command: " + line + "\n");
+    }
+}
+
+void handle_cli()
+{
+    SWSS_LOG_ENTER();
+
+    char buffer[BUFFER_SIZE];
+
+    run_user = true;
+
+    while (run_user)
+    {
+        if (!sendtoclient(CLI_PROMPT))
+        {
+            return;
+        }
+
+        // blocking
+        ssize_t n = read(cli_client_socket, buffer, BUFFER_SIZE - 1);
+
+        if (n == 0)
+        {
+            SWSS_LOG_NOTICE("stream closed");
+            return;
+        }
+
+        if (n < 0)
+        {
+            if (run_cli || run_user)
+            {
+                SWSS_LOG_ERROR("read failed, errno: %s", strerror(errno));
+            }
+
+            return;
+        }
+
+        const std::string& s = std::string(buffer, n);
+
+        handle_cli_commands(s);
```

Comment:
>handle_cli_commands [](start = 8, length = 19)

Is it possible that s contains multiple line? Considering redirect input from a file or pipe.